slides/r-adoption-jan2023.qmd

```diff
@@ -0,0 +1,204 @@
+
+## Idea
+
+-   Because `glmmTMB` is always using a random effects representation, we cannot have a real unstructured model (uses $\sigma = \varepsilon > 0$ trick)
+-   We only want to fit a fixed effects model with a structured covariance matrix for each subject
+-   The idea is then to use the Template Model Builder (`TMB`) directly - as it is also underlying `glmmTMB` - but code the exact model we want
+-   We do this by implementing the log-likelihood in `C++` using the `TMB` provided libraries
+
+## Advantages of `TMB`
+
+-   Fast `C++` framework for defining objective functions (`Rcpp` would have been alternative interface)
+-   Automatic differentiation of the log-likelihood as a function of the variance parameters
+-   We get the gradient and Hessian exactly and without additional coding
+-   This can be used from the R side with the `TMB` interface and plugged into optimizers
+
+## What's next with {mmrm} and SWE WG
+
+-   Prepare public training materials to disseminate best practice for software engineering in the Biostatistics community
+    -   At the beginning of February, a face-to-face workshop will take place in Basel, Switzerland with a focus on open-source software for clinical trials
+    -   Organize conference sessions with a focus on statistical software engineering at CEN, JSM and ASA/FDA Workshop
+    -   Video clips in preparation with a focus on best practices for software engineering
+
+## Why it's not just another package
+
+-   Ongoing maintenance and support from the pharmaceutical industry that is supported by American Statistical Association
+-   Package is part of the mission, but to emphasize our goal is to push out information on practices for engineering high-quality open-source statistical software
+
+## Long term perspective
+
+-   Software engineering is a critical competence in producing high-quality statistical software
+-   A lot of work need to be done regarding the establishment, dissemination and adoption of best practices for engineering open-source software
+-   Improving the way software engineering is done will help improve the efficiency, reliability and innovation within Biostatistics
+
+## Comparing SAS and R
+
+- Estimation Methods
+- Covariance structures
+- Degrees of Freedom
+- contrasts
+- comparing estimates
+- run times
+
+## Estimation Methods
+
+|       Method     | mmrm::mmrm | SAS |
+|:----------------|:----------:|:---:|
+|       ML         |     X      |  X  |
+|       REML       |     X      |  X  |
+|       MIVQUE0*   |            |  X  |
+
+*MIVQUE0: minimum variance quadratic unbiased estimation of the covariance parameters
+
+##  Covariance structures
+
+- SAS has 23 non-spatial covariance structures, while mmrm has 10. 
+  - 9 structures intersect with SAS
+  - Ante-dependence (homogeneous) is only in the mmrm package. 
+- SAS has 14 spatial covariance structures compared to the spatial exponential one available in mmrm.
```

Comment:
you could add here that the package is easily extensible, in particular for the more often used non-spatial structures. so if anybody needs additional one, let us know.

Reply:
is there a mechanism to bring your own struct in session? or would it be through a feature request?

Reply:
Through feature request